chall05/rjaakonm.rb

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env ruby
+# ruby version 2.5.2.3
+
+require 'open-uri'
+
+$array = Array.new
+$counter = 0
+
+def getArticleSource(string)
+	url = 'https://en.wikipedia.org/wiki/'
+	source = open(url + string).read
+    rescue
+		puts 'no wikipedia page for this word!'
+        exit
+    else
+    	GetFirstLink(source)
+end
+
+def GetFirstLink(source)
+	# Get content part of article
+	pattern = /<p>[\s\S]+?<\/p>/
+	paragraphs = source.scan(pattern)
+	if !paragraphs[0]
+		puts 'no wikipedia page for this word!'
+		exit
+	end
+
+    # Get wikipedia links from content and loop to find first link
+	pattern = /a href="\/wiki\/([\s\S]*?)["]/
+    paragraphs.each{ |paragraph|
+		links = paragraph.scan(pattern)
+        links.each{ |link|
+			if (link[0] == 'Philosophy')
+				puts '!!! Reach Philosophy !!!'
+				exit
+			elsif ($array[0] and $array.include? link[0])
```

Comment:
I suggest avoiding writing your params that way even If ruby allows it. It's better to keep consistent with other languages you develop or will develop with.